be/src/exec/vectorized/hdfs_scan_node.cpp

```diff
@@ -578,6 +586,31 @@ void HdfsScanNode::_init_partition_expr_map() {
     }
 }
 
+void HdfsScanNode::_init_hudi_partition_expr_map() {
+    if (_scan_ranges.empty()) {
+        return;
+    }
+
+    for (auto& scan_range : _scan_ranges) {
+        auto* partition_desc = _hudi_table->get_partition(scan_range.partition_id);
+
+        if (_partition_values_map.find(scan_range.partition_id) == _partition_values_map.end()) {
+            _partition_values_map[scan_range.partition_id] = partition_desc->partition_key_value_evals();
+        }
+    }
+
+    if (_has_partition_columns && _has_partition_conjuncts) {
+        auto it = _partition_values_map.begin();
+        while (it != _partition_values_map.end()) {
+            if (_filter_partition(it->second)) {
+                _partition_values_map.erase(it++);
```

Comment:
prefer `it = _partition_values_map.erase(it);` since c++11 returns the iterator following the last removed element.

Reply:
done